### PR TITLE
#78: addDocuments and updateDocuments should take pk as optional param

### LINKED ISSUE #78: The addDocuments and updateDocuments methods should take a primary key as optional parameter
The `addDocument` and `updateDocuments` methods do not take any primary key as parameter, but they should.
Related docs:
- https://docs.meilisearch.com/references/documents.html#add-or-replace-documents
- https://docs.meilisearch.com/references/documents.html#add-or-update-documents

⚠️ The primary key parameter should be optional!

- [ ] add in the code base
- [ ] add tests

### PR DESCRIPTION
**Pull request by vishnugt**
Fixes #78 
cc: @alallema

### CODE REVIEW

File: src/main/java/com/meilisearch/sdk/Documents.java [hunk 1]
```diff
@@ -54,11 +54,15 @@ String getDocuments(String uid, int limit) throws Exception {
      *
      * @param uid Partial index identifier for the document
      * @param document String containing the document to add
+     * @param identifier Identifier of the document
```
Comment (alallema) [inline src/main/java/com/meilisearch/sdk/Documents.java:57]:
```suggestion
     * @param primaryKey PrimaryKey of the document
```

File: src/main/java/com/meilisearch/sdk/Documents.java [hunk 2]
```diff
@@ -54,11 +54,15 @@ String getDocuments(String uid, int limit) throws Exception {
...
      * @return String containing the added document
      * @throws Exception if the client request causes an error
      */
-    String addDocuments(String uid, String document) throws Exception {
+    String addDocuments(String uid, String document, String identifier) throws Exception {
```
Comment (alallema) [inline src/main/java/com/meilisearch/sdk/Documents.java:61]:
```suggestion
    String addDocuments(String uid, String document, String primaryKey) throws Exception {
```

File: src/main/java/com/meilisearch/sdk/Documents.java [hunk 3]
```diff
@@ -54,11 +54,15 @@ String getDocuments(String uid, int limit) throws Exception {
...
         String requestQuery = "/indexes/" + uid + "/documents";
+        if (identifier != null) {
```
Comment (alallema) [inline src/main/java/com/meilisearch/sdk/Documents.java:63]:
```suggestion
        if (primaryKey != null) {
```

File: src/main/java/com/meilisearch/sdk/Documents.java [hunk 4]
```diff
@@ -54,11 +54,15 @@ String getDocuments(String uid, int limit) throws Exception {
...
+            requestQuery += "?primaryKey=" + identifier;
```
Comment (alallema) [inline src/main/java/com/meilisearch/sdk/Documents.java:64]:
```suggestion
            requestQuery += "?primaryKey=" + primaryKey;
```

File: src/main/java/com/meilisearch/sdk/Documents.java [hunk 5]
```diff
@@ -67,11 +71,15 @@ String addDocuments(String uid, String document) throws Exception {
      *
      * @param uid Partial index identifier for the document
      * @param document String containing the document to replace the existing document
+     * @param identifier Identifier of the document
```
Comment (alallema) [inline src/main/java/com/meilisearch/sdk/Documents.java:74]:
```suggestion
     * @param primaryKey PrimaryKey of the document
```

File: src/main/java/com/meilisearch/sdk/Documents.java [hunk 6]
```diff
@@ -67,11 +71,15 @@ String addDocuments(String uid, String document) throws Exception {
...
      * @return String containing the added document
      * @throws Exception if the client request causes an error
      */
-    String updateDocuments(String uid, String document) throws Exception {
+    String updateDocuments(String uid, String document, String identifier) throws Exception {
```
Comment (alallema) [inline src/main/java/com/meilisearch/sdk/Documents.java:78]:
```suggestion
    String updateDocuments(String uid, String document, String primaryKey) throws Exception {
```

File: src/main/java/com/meilisearch/sdk/Documents.java [hunk 7]
```diff
@@ -67,11 +71,15 @@ String addDocuments(String uid, String document) throws Exception {
...
         String requestQuery = "/indexes/" + uid + "/documents";
+        if (identifier != null) {
+            requestQuery += "?primaryKey=" + identifier;
```
Comment (alallema) [inline src/main/java/com/meilisearch/sdk/Documents.java:81]:
```suggestion
        if (primaryKey != null) {
            requestQuery += "?primaryKey=" + primaryKey;
```

File: src/main/java/com/meilisearch/sdk/Index.java [hunk 8]
```diff
@@ -84,7 +84,19 @@ public String getDocuments(int limits) throws Exception {
...
     public String addDocuments(String document) throws Exception {
-        return this.documents.addDocuments(this.uid, document);
+        return this.documents.addDocuments(this.uid, document, null);
+    }
+
+    /**
+     * Adds a document in the index
+     *
+     * @param document Document to add in JSON string format
+     * @param identifier Identifier of the document to add
```
Comment (alallema) [inline src/main/java/com/meilisearch/sdk/Index.java:94]:
```suggestion
     * @param primaryKey PrimaryKey of the document to add
```

File: src/main/java/com/meilisearch/sdk/Index.java [hunk 9]
```diff
@@ -84,7 +84,19 @@ public String getDocuments(int limits) throws Exception {
...
+     * @return MeiliSearch API response
+     * @throws Exception if something goes wrong
+     */
+    public String addDocuments(String document, String identifier) throws Exception {
```
Comment (alallema) [inline src/main/java/com/meilisearch/sdk/Index.java:98]:
```suggestion
    public String addDocuments(String document, String primaryKey) throws Exception {
```

File: src/main/java/com/meilisearch/sdk/Index.java [hunk 10]
```diff
@@ -84,7 +84,19 @@ public String getDocuments(int limits) throws Exception {
...
+        return this.documents.addDocuments(this.uid, document, identifier);
```
Comment (alallema) [inline src/main/java/com/meilisearch/sdk/Index.java:99]:
```suggestion
        return this.documents.addDocuments(this.uid, document, primaryKey);
```

File: src/main/java/com/meilisearch/sdk/Index.java [hunk 11]
```diff
@@ -95,7 +107,19 @@ public String addDocuments(String document) throws Exception {
...
     public String updateDocuments(String document) throws Exception {
-        return this.documents.updateDocuments(this.uid, document);
+        return this.documents.updateDocuments(this.uid, document, null);
+    }
+
+    /**
+     * Updates a document in the index
+     *
+     * @param document Document to update in JSON string format
+     * @param identifier Identifier of the document
```
Comment (alallema) [inline src/main/java/com/meilisearch/sdk/Index.java:117]:
```suggestion
     * @param primaryKey PrimaryKey of the document
```

File: src/main/java/com/meilisearch/sdk/Index.java [hunk 12]
```diff
@@ -95,7 +107,19 @@ public String addDocuments(String document) throws Exception {
...
+     * @return MeiliSearch API response
+     * @throws Exception if something goes wrong
+     */
+    public String updateDocuments(String document, String identifier) throws Exception {
+        return this.documents.updateDocuments(this.uid, document, identifier);
```
Comment (alallema) [inline src/main/java/com/meilisearch/sdk/Index.java:122]:
```suggestion
    public String updateDocuments(String document, String primaryKey) throws Exception {
        return this.documents.updateDocuments(this.uid, document, primaryKey);
```

File: src/test/java/com/meilisearch/integration/DocumentsTest.java [hunk 13]
```diff
@@ -62,6 +63,46 @@ public void testAddDocumentsSingle() throws Exception {
         assertEquals("Drama", movies[0].getGenres()[1]);
     }
 
+    /** Test add Documents with identifier */
+    @Test
+    public void testAddDocumentsWithSuppliedIdentifier() throws Exception {
```
Comment (alallema) [inline src/test/java/com/meilisearch/integration/DocumentsTest.java:68]:
```suggestion
    public void testAddDocumentsWithSuppliedPrimaryKey() throws Exception {
```

File: src/test/java/com/meilisearch/integration/DocumentsTest.java [hunk 14]
```diff
@@ -62,6 +63,46 @@ public void testAddDocumentsSingle() throws Exception {
...
+
+        String indexUid = "TestAddWithIdentifier";
```
Comment (alallema) [inline src/test/java/com/meilisearch/integration/DocumentsTest.java:70]:
```suggestion
        String indexUid = "TestAddWithPrimaryKey";
```

**Review by alallema — COMMENTED** [review]
Great Job! Thank you so much for this PR @vishnugt.
I just change some naming.

**Comment by vishnugt** [comment]
@alallema  Thanks for the review, I've updated all instances of `identifier` to `primaryKey`

**Comment by alallema** [comment]
Oppsy I click on `Update Branch` to quickly

**Review by alallema — APPROVED** [review]
LGTM!
Thank again @vishnugt! Don't forget to complete [this form](https://meilisearch.typeform.com/to/rPexR5gq) if you haven't done so already.

**Comment by alallema** [comment]
bors merge

**Pull request merged**